### PR TITLE
fix(odata-service-inq): handle auth errors for catalog requests

### PR DESCRIPTION
**Pull request by cianmSAP**
#3116 

If the V2 request returns an auth error, we run the V4 catalog request
However if this also fails, we should return the original auth error so it can be handled accordingly

### CODE REVIEW

File: packages/odata-service-inquirer/src/prompts/connectionValidator.ts [hunk 1]
```diff
@@ -438,34 +438,40 @@ export class ConnectionValidator {
...
         }
     }
 
     /**
-     * Check if we should attempt to use the v4 catalog service as a fallback.
+     * Depending on the initial error, it will run the catalog request and handle the error if it occurs.
      *
-     * @param statusCode http status code, if not provided will return false as we cannot determine the reason for v2 catalog request failure
-     * @returns true if we should attempt the v4 catalog service
+     * @param error - error
```
Comment (IainSAP) [inline packages/odata-service-inquirer/src/prompts/connectionValidator.ts:448]: Determine if a v4 catalog request should be made based on the specified error and if the error originated from a v2 or v4 catalog request. If the error originated from a v2 catalog request and it is of type 401/403/404 then we try the v4 catalog. If the v4 catalog request fails, for any reason, and the v2 catalog request failed with 401/403 we will throw that error, ultimately resulting in a basic auth prompt for the end-user. Otherwise the v4 catalog request is thrown, which may also be a 401/403, again resulting in a basic auth prompt and re-validation. If both catalogs have returned 404 nothing else can be done, the error is thrown and reported.

@param error An error returned from either a v2 or v4 catalog `listServices` request.

Comment (cianmSAP) [reply packages/odata-service-inquirer/src/prompts/connectionValidator.ts]: Updated, thank you

**Review by IainSAP — DISMISSED** [review]
Thanks @cianmSAP 

- Bug fix seems appropriate to cover these edge cases 
- Test added to cover

**Review by longieirl — DISMISSED** [review]
Changes look good.
Changeset is good.
Tests to cover change are good.

**Comment by IainSAP** [comment]
@cianmSAP fyi: This needs to go ASAP: https://github.com/SAP/open-ux-tools/pull/3120

**Review by IainSAP — DISMISSED** [review]
Re-approving...but there is another pending VH fix.

**Review by IainSAP — APPROVED** [review]
Re-approving

**Review by longieirl — APPROVED** [review]
Re-approving
